x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -1029,6 +1030,114 @@ export class AlertsClient {
     }
   }
 
+  /**
+   * Performs a `find` query to extract aggregations on alert groups
+   */
+  public getGroupAggregations({
+    featureIds,
+    groupByField,
+    aggregations,
+    filters,
+    pageIndex,
+    pageSize,
+    sort = [{ unitsCount: { order: 'desc' } }],
+  }: {
+    /**
+     * The feature ids the alerts belong to, used for authorization
+     */
+    featureIds: string[];
+    /**
+     * The field to group by
+     * @example "kibana.alert.rule.name"
+     */
+    groupByField: string;
+    /**
+     * The aggregations to perform on the groupByField buckets
+     */
+    aggregations?: TypeOf<typeof alertsAggregationsSchema>;
+    /**
+     * The filters to apply to the query
+     */
+    filters?: estypes.QueryDslQueryContainer[];
+    /**
+     * Any sort options to apply to the groupByField aggregations
+     */
+    sort?: SortCombinations[];
+    /**
+     * The page index to start from
+     */
+    pageIndex: number;
+    /**
+     * The page size
+     */
+    pageSize: number;
+  }) {
+    const uniqueValue = uuidv4();
+    if (pageIndex > MAX_ALERTS_PAGES) {
+      throw Boom.badRequest(
+        `The provided pageIndex value is too high. The maximum allowed pageIndex value is ${MAX_ALERTS_PAGES}.`
+      );
+    }
+    if (Math.max(pageIndex, pageIndex * pageSize) > MAX_PAGINATED_ALERTS) {
+      throw Boom.badRequest(
+        `The number of documents is too high. Paginating through more than ${MAX_PAGINATED_ALERTS} documents is not possible.`
+      );
+    }
+    return this.find({
+      featureIds,
+      aggs: {
+        groupByField: {
+          terms: {
+            field: 'groupByField',
+            size: MAX_ALERTS_GROUPING_QUERY_SIZE,
+          },
+          aggs: {
+            unitsCount: { value_count: { field: 'groupByField' } },
+            bucket_truncate: {
+              bucket_sort: {
+                sort,
+                from: pageIndex * pageSize,
+                size: pageSize,
+              },
+            },
+            ...(aggregations ?? {}),
+          },
+        },
+        unitsCount: { value_count: { field: 'groupByField' } },
+        groupsCount: { cardinality: { field: 'groupByField' } },
+      },
+      query: {
+        bool: {
+          filter: filters,
+        },
+      },
+      runtimeMappings: {
+        groupByField: {
+          type: 'keyword',
+          script: {
+            source:
+              // When size()==0, emits a uniqueValue as the value to represent this group  else join by uniqueValue.
+              "if (doc[params['selectedGroup']].size()==0) { emit(params['uniqueValue']) }" +
```

Comment:
nitpick: can use `` to avoid concatenating the strings

Reply:
Yep, I though it was worth keeping it as it was originally written by Security though, to comment the two parts separately 🙂